File: PosHubApi/Dtos/ModifierDto.cs
using System.Text.Json.Serialization;

namespace PosHubApi.Dtos
{
    public class ModifierDto
    {
        [JsonPropertyName("catalogModifierId")]
        public string PosReference { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string PosVersion { get; set; }
        public string OriginalImageUrl { get; set; }
        public decimal Price { get; set; }
        public decimal? InStorePrice { get; set; }
        public decimal TaxRate { get; set; }
        public bool IsTaxIncluded { get; set; } = true;
        public bool ContainsAlcohol { get; set; }
        public bool ContainsTobacco { get; set; }
        public bool IsBikeFriendly { get; set; }
        public bool ShowOnline { get; set; }
        public int Position { get; set; }
        public int MinPermitted { get; set; }
        public int MaxPermitted { get; set; }
        public NutritionalInfoDto NutritionalInfo { get; set; } = new();
        public List<SelectionDto> Selections { get; set; } = new();

    }
}

File: poshub-spa/src/app/dtos/modifierDto.ts
export class ModifierDto {
   constructor(
  public posReference: string = '',
  public name: string = '',
  public description: string = '',
  public posVersion: string = '',
  public originalImageUrl: string = '',
  public price: number = 0,
  public inStorePrice?: number,
  public taxRate: number = 0,
  public isTaxIncluded: boolean = true,
  public containsAlcohol: boolean = false,
  public containsTobacco: boolean = false,
  public isBikeFriendly: boolean = false,
  public showOnline: boolean = false,
  public position: number = 0,
  public minPermitted: number = 0,
  public maxPermitted: number = 0,
    //public nutritionalInfo: NutritionalInfoDto = new NutritionalInfoDto(),
   // public selections: SelectionDto[] = []
  public accountId: string = '',
  public createdAt: Date = new Date(),
  public taxRateIds: string[] = [],
  public locationId: string = '',
  public imageUrl: string = '',
  public updatedAt: Date = new Date()
     ) {}
}
